modules/fedimint-wallet/src/bitcoincore_rpc.rs

```diff
@@ -50,6 +52,11 @@ impl bitcoincore_rpc::RpcApi for RetryClient {
                     break ret;
                 }
                 Err(e) => {
+                    // Do not retry if transaction was alreadty submitted
+                    if let Error::JsonRpc(JsonError::Rpc(RpcError { code: -27, .. })) = e {
```

Comment:
Can we be sure that for no other rpc call `code = -27 means something?

It seems to me it might be safer to to handle this on line 115, and return `Ok`. After all - the job of the function was to make sure tx is submitted. If it was already submitted - it should be a success? Alternatively change to `Result<Option<()>` or `Result<boo>`.

Reply:
Afaik there's no spec, but the error codes don't change:
https://github.com/bitcoin/bitcoin/blob/0.21/src/rpc/protocol.h#L48

The error will still get propagated to the user (`panics` like all other errors), just no retries will be performed.

I'm not sure how to handle on line 115, because that line is actually delegating to the `call` fn that does the retry.

Reply:
In the lambda, no?

```rust
if let Err(error) =
            fedimint_api::task::block_in_place(|| {
                  match  self.send_raw_transaction(&transaction) {
                    Err({ code: 13, ...}) => return Ok(()),
                    other => other,
                  }

       })
        {
            warn!(?error, "Submitting transaction failed");
}
```

Something like this, except correct. :D

Reply:
Sorry. I've confused `call` with `block_in_place` there. We would have to refactor this code a bit.